internal/providers/pluginfw/resources/clusters/resource_library.go

```diff
@@ -0,0 +1,237 @@
+	"strings"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/terraform-provider-databricks/clusters"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/compute_tf"
+	"github.com/databricks/terraform-provider-databricks/libraries"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/databricks/databricks-sdk-go"
+)
+
+const libraryDefaultInstallationTimeout = 15 * time.Minute
+
+var _ resource.ResourceWithConfigure = &LibraryResource{}
+
+func ResourceLibrary() resource.Resource {
+	return &LibraryResource{}
+}
+
+func parseId(id string) (string, string) {
+	split := strings.SplitN(id, "/", 2)
+	if len(split) != 2 {
+		return "unknown", "unknown"
+	}
+	return split[0], split[1]
+}
+
+func readLibrary(ctx context.Context, w *databricks.WorkspaceClient, waitParams compute.Wait, libraryRep string, libraryExtended *LibraryExtended) diag.Diagnostics {
+	res, err := libraries.WaitForLibrariesInstalledSdk(ctx, w, waitParams, libraryDefaultInstallationTimeout)
+	if err != nil {
+		return diag.Diagnostics{diag.NewErrorDiagnostic("failed to wait for library installation", err.Error())}
+	}
+
+	for _, v := range res.LibraryStatuses {
+		thisRep := v.Library.String()
+		if thisRep == libraryRep {
+
+			diags := converters.GoSdkToTfSdkStruct(ctx, v.Library, libraryExtended)
+
+			if diags.HasError() {
+				return diags
+			}
+
+			libraryExtended.ClusterId = types.StringValue(waitParams.ClusterID)
+
+			return nil
+		}
+	}
+	return diag.Diagnostics{diag.NewErrorDiagnostic("failed to find the installed library", fmt.Sprintf("failed to find %s on %s", libraryRep, waitParams.ClusterID))}
+}
+
+type LibraryExtended struct {
+	compute_tf.Library
+	ClusterId types.String `tfsdk:"cluster_id" tf:""`
+}
+
+type LibraryResource struct {
+	Client *common.DatabricksClient
+}
+
+func (r *LibraryResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "databricks_library_pluginframework"
+}
+
+func (r *LibraryResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Library",
+		Attributes: tfschema.ResourceStructToSchemaMap(LibraryExtended{}, func(c tfschema.CustomizableSchema) tfschema.CustomizableSchema {
```

Comment:
Do we need to modify the plan to set all library attributes to the equivalent of ForceNew? Today, this is what is done by `common.Resource` for resources without update defined:
```
		// set ForceNew to all attributes with CRD
		queue := []*schema.Resource{
			{Schema: r.Schema},
		}
		for {
			head := queue[0]
			queue = queue[1:]
			for _, v := range head.Schema {
				if v.Computed {
					continue
				}
				if nested, ok := v.Elem.(*schema.Resource); ok {
					queue = append(queue, nested)
				}
				v.ForceNew = true
			}
			if len(queue) == 0 {
				break
			}
		}
```
I believe all that is needed is to set `<type>planmodifier.RequiresReplace()` plan modifier for all top-level fields.

Let's validate this behavior with an integration test as well.